src/components/CreateOrUpdateEvent.svelte

```diff
@@ -0,0 +1,176 @@
+<script>
+  import LocationInput from 'components/LocationInput.svelte'
+  import Uploader from 'components/Uploader.svelte'
+  import { throwError } from 'data/error.js'
+  import { events, update, create } from 'data/events.js'
+  import { me } from 'data/user.js'
+
+  import { format, addDays } from 'date-fns'
+  import { goto, params } from '@roxi/routify'
+
+  const tomorrow = format(addDays(Date.now(), 1), 'yyyy-MM-dd')
+
+  $: event = {}
+  $: existingEvent = $events.find(({ id }) => id === $params.eventId)
+  $: initializeUpdates(existingEvent)
+  $: isNew = !event.id
+
+  new Date().getDate
+
+  const parseDate = dateString => {
+    let date = new Date(dateString)
+    let month = date.getMonth() + 1
+    let dayOfMonth = date.getDate() + 1
+    let monthLength = `${month}`.length
+    let dateLength = `${dayOfMonth}`.length
+    return `${date.getFullYear()}-${monthLength == 1 ? `0${month}` : month}-${dateLength == 1 ? `0${dayOfMonth}` : dayOfMonth}`
+  }
+
+  function initializeUpdates(eventBeingEdited) {
+    if (!eventBeingEdited) return
+    event = Object.assign({}, eventBeingEdited)
+    event.start_date = parseDate(event.start_date)
+    event.end_date = parseDate(event.end_date)
+  }
+
+  function assertHas(value, errorMessage) {
+    if (!value) {
+      throwError(errorMessage)
+    }
+  }
+
+  function validate(event) {
+    assertHas(event.name, 'Please provide the name of the event')
+    assertHas(event.location, 'Please provide a location')
+    assertHas(event.start_date, 'Please tell us the start date')
+    assertHas(event.end_date, 'Please tell us the end date')
+  }
+
+  async function onSubmit() {
+    validate(event)
+
+    if (isNew) await create(event)
+    else await update(event)
+
+    $goto(`/events`)
+
+  }
+
+  function imageUploaded(e) {
+    event.image_file = e.detail
+  }
+
+  function onLocationChanged(e) {
+    event.location = e.detail
+  }
+
+</script>
+
+<style>
+  .preview {
+    max-height: 160px;
+  }
+</style>
+  
+{#if $events.length != 0 && !existingEvent && !isNew }
+  <h2 class="mb-e">Event does not exist!</h2>photo
+{:else if existingEvent && event.created_by?.id != $me.id && !isNew }
+  <h2 class="mb-e">You cannot edit this event!</h2>
+{:else if $events.length != 0 }
```

Comment:
```suggestion
{:else}
```
This should still render even if there are no events

Reply:
I had that (now `{:else if $events.length }` on Mr. Wilson's suggestion) because there is a brief instant where the $events variable hasn't been initialized (still fetching from the store), and in that brief moment, if this code wasn't there, you could see the form before, for instance, the "Event does not exist!" message displays or "You cannot edit this event!".
Basically just makes the UI look cleaner in my opinion.

Reply:
That's a good observation, and I agree that it's an important thing to address. Will it still work in the case where there are no events? Hopefully that doesn't happen in real life, but it might because it only includes future events.

Reply:
Yes, I'm pretty sure. Although, before (when it was `{:else if $events.length != 0 }`) we would've had a problem. But now I think it's fine because I'm pretty sure if there were no events the `$events` variable would just be an empty store instead of an uninitialized variable. I'll go ahead and double-check that.

Reply:
Oh, wait. Scratch that, it doesn't work if there are no events. Let we work at it for a bit.

Reply:
You might need to reference `loading` from `events.js`

Reply:
What exactly does that do?

Reply:
It is `true` while there is a pending API call for loading the events store.